ksqldb-udf/src/main/java/io/confluent/ksql/schema/ksql/types/SqlLambda.java

```diff
@@ -15,47 +15,36 @@
 
 package io.confluent.ksql.schema.ksql.types;
 
-import static java.util.Objects.requireNonNull;
-
-import com.google.common.collect.ImmutableList;
 import com.google.errorprone.annotations.Immutable;
 import io.confluent.ksql.schema.utils.FormatOptions;
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
 
 /**
- * An internal object to track the input types and return types for a lambda
- * argument that's seen as an argument inside of a UDF.
+ * An internal object to represent a lambda function, we can't
  */
 @Immutable
-public final class SqlLambda {
+public class SqlLambda {
 
-  private final List<SqlType> inputTypes;
-  private final SqlType returnType;
+  private final int numInputs;
 
   public static SqlLambda of(
-      final List<SqlType> inputType,
-      final SqlType returnType
+      final Integer numInputs
```

Comment:
Java doesn't recognize that the parameters of the bifunction are different, if we were to write the UDF `foo` and try to have those two variants of the function, Java would complain since they have the same erasure. That's why we're able to just check the num inputs when getting the correct function from UDF factory

Reply:
Ah right my bad I forgot it just gets turned into BiFunction at the end of the day.